src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationExtensions.cs

```diff
@@ -81,5 +81,31 @@ public static bool Exists(this IConfigurationSection section)
             }
             return section.Value != null || section.GetChildren().Any();
         }
+
+        /// <summary>
+        /// Gets a configuration sub-section with the specified key.
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <param name="key">The key of the configuration section.</param>
+        /// <returns>The <see cref="IConfigurationSection"/>.</returns>
+        /// <remarks>
+        ///     If no matching sub-section is found with the specified key, an exception is raised.
+        /// </remarks>
+        /// <exception cref="System.InvalidOperationException">There is no section with key <paramref name="key"/>.</exception>
+        public static IConfigurationSection GetRequiredSection(this IConfiguration configuration, string key)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
```

Comment:
You could also perhaps add a test covering this use case. (configuration is null)

Reply:
Hi @maryamariyan 
I added this one :)